optuna/trial.py

```diff
@@ -1,6 +1,8 @@
 import abc
 from datetime import datetime
+from datetime import timedelta
```

Comment:
`timedelta` should be imported in `TYPE_CHECKING` statement. Rather, I'd not import `timedelta`.

Let me know should you have any feelings/reasons 🙂

Reply:
I agree with you. I will make it imported in `TYPE_CHECKING`. Thank you for your comment.